rising/transforms/compose.py

```diff
@@ -99,12 +97,82 @@ def forward(self, *seq_like, **map_like) -> Union[Sequence, Mapping]:
             dict with transformed data
         """
         assert not (seq_like and map_like)
+        assert len(self.transforms) == len(self.transform_order)
         data = seq_like if seq_like else map_like
 
-        for trafo in self.transforms:
-            data = self.transform_call(data, trafo)
+        if self.shuffle:
+            shuffle(self.transform_order)
+
+        for idx in self.transform_order:
+            data = self.transform_call(data, self.transforms[idx])
         return data
 
+    @property
+    def transforms(self) -> torch.nn.ModuleList:
+        """
+        Transforms getter
+
+        Returns
+        -------
+        torch.nn.ModuleList
+            transforms to compose
+        """
+        return self._transforms
+
+    @transforms.setter
+    def transforms(self, transforms: Union[AbstractTransform,
+                                           Sequence[AbstractTransform]]):
+        """
+        Transforms setter
+
+        Parameters
+        ----------
+        transforms: Union[AbstractTransform, Sequence[AbstractTransform]]
+            one or multiple transformations which are applied in consecutive
+            order
+
+        Returns
+        -------
+        torch.nn.ModuleList
+            transforms to compose
+        """
+        # make transforms a list to be mutable.
+        # Otherwise the enforced typesetting below might fail.
+        if isinstance(transforms, tuple):
+            transforms = list(transforms)
+
+        for idx, trafo in enumerate(transforms):
+            if not isinstance(trafo, torch.nn.Module):
+                transforms[idx] = _TransformWrapper(trafo)
+
+        self._transforms = torch.nn.ModuleList(transforms)
+        self.transform_order = list(range(len(self.transforms)))
+
+    @property
+    def shuffle(self) -> bool:
+        """
+        Getter for attribute shuffle
+
+        Returns
+        -------
+        bool
+            True if shuffle is enabled, False otherwise
+        """
+        return self._shuffle
+
+    @shuffle.setter
+    def shuffle(self, shuffle: bool):
+        """
+        Setter for shuffle
+
+        Parameters
+        ----------
+        shuffle : bool
+            new status of shuffle
+        """
+        self._shuffle = shuffle
+        self.transform_order = list(range(len(self.transforms)))
```

Comment:
Can we only create this if shuffle is set to true?

Reply:
The current implementation always uses this right now to avoid switching between different for loops inside the forward. In case `shuffle=false` its just the regular ordering. I added this as a property in case someone changes shuffle from `true` to `which` would lead to the wrong behaviour (because the indices are shuffled in-place) and thus needs  a reset.